features/dd-sdk-android-flags/src/main/kotlin/com/datadog/android/flags/featureflags/internal/persistence/FlagsStateDeserializer.kt

```diff
@@ -0,0 +1,121 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.flags.featureflags.internal.persistence
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.core.internal.persistence.Deserializer
+import com.datadog.android.flags.featureflags.internal.model.FlagsStateEntry
+import com.datadog.android.flags.featureflags.internal.model.JsonKeys
+import com.datadog.android.flags.featureflags.internal.model.PrecomputedFlag
+import com.datadog.android.flags.featureflags.model.EvaluationContext
+import org.json.JSONException
+import org.json.JSONObject
+
+internal class FlagsStateDeserializer(
+    private val internalLogger: InternalLogger
+) : Deserializer<String, FlagsStateEntry> {
+
+    override fun deserialize(model: String): FlagsStateEntry? {
+        return try {
+            val json = JSONObject(model)
+
+            @Suppress("UnsafeThirdPartyFunctionCall") // JSONObject operations wrapped in try-catch
+            val contextJson = json.getJSONObject(JsonKeys.EVALUATION_CONTEXT.value)
+            val evaluationContext = deserializeEvaluationContext(contextJson)
+
+            @Suppress("UnsafeThirdPartyFunctionCall") // JSONObject operations wrapped in try-catch
+            val flagsJson = json.getJSONObject(JsonKeys.FLAGS.value)
+            val flags = deserializeFlags(flagsJson)
+
+            @Suppress("UnsafeThirdPartyFunctionCall") // JSONObject operations wrapped in try-catch
+            val timestamp = json.getLong(JsonKeys.LAST_UPDATE_TIMESTAMP.value)
+
+            FlagsStateEntry(
+                evaluationContext = evaluationContext,
+                flags = flags,
+                lastUpdateTimestamp = timestamp
+            )
+        } catch (e: JSONException) {
+            internalLogger.log(
+                InternalLogger.Level.ERROR,
+                InternalLogger.Target.MAINTAINER,
+                { "Failed to deserialize FlagsStateEntry from JSON" },
+                e
+            )
+            null
+        }
+    }
+
+    @Suppress("UnsafeThirdPartyFunctionCall") // JSONObject operations wrapped in try-catch
+    private fun deserializeEvaluationContext(contextJson: JSONObject): EvaluationContext {
+        val targetingKey = contextJson.getString(JsonKeys.TARGETING_KEY.value)
+        val attributesJson = try {
```

Comment:
it can be slightly simplified:

```
val attributes = try {
  deserializeAttributes(contextJson.getJSONObject(JsonKeys.ATTRIBUTES.value))
} catch {
  emptyMap()
}
```

and then you don't need to have nullable type in `deserializeAttributes` and do null-check there.